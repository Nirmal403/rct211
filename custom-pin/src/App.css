.App {
  text-align: center;
  background-color: yellow;
  height: 100vh;
  background: url("https://thumbs.dreamstime.com/z/otp-one-time-password-acronym-technology-concept-background-226004658.jpg") no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}
.otp{
  color:red;
  font-style: italic;
}
.App input{
  height: 40px;
  width: 40px;
  margin: 1rem 0.5rem;
  text-align: center;
  color: red;
  font-weight: 600;
  border: 1px solid blue;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
