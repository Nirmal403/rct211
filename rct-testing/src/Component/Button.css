.button{
    color: white;
    margin-top:1rem;
    cursor: pointer;
    padding: 10px;
    border-radius:5px;
}
.red{
    background-color: red;
}
.blue{
    background-color: blue;
}
.green{
    background-color: greenyellow;
}

.ghost{
    border: none;
}
.outline{
    border: 1px solid green;
    color: black;
}