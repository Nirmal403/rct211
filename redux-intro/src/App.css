.App {
  text-align: center;
  background-color: yellow;
  height: 100vh;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}
.decrement{
  background-color: red;
  color: white;
  font-size: 25px;
  width: 50px;
  height: 50px;
  border: 1px solid red;
  margin-left: 5px;
  border-radius: 5px;

}
.increment{
  background-color: green;
  border: 1px solid green;
  font-size: 25px;
  margin-left: 5px;
  border-radius: 5px;
  width: 50px;
  height: 50px;
  color: white;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
