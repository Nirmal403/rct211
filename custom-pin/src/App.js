import { useState } from 'react';
import './App.css';
import Pin from './Components/Pin';

function App() {
  const [otp, setOtp] = useState("");
  const [otpValid,setOtpValid] =useState(false);
  return (
    <div className="App">
     <Pin length={5}
     setOtp={setOtp}
     isOtpValid={otpValid}/>
     <div>
      <h3>The OTP input is: <span className="otp">{otp}</span></h3>
     </div>
    </div>
  );
}

export default App;
