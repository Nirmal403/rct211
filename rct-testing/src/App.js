import { useState } from 'react';
import './App.css';
import Button from './Component/Button';

function App() {
  const [theme, setTheme] = useState('light');
  return (
    <div className="App">
      <h2> Theme is {theme}</h2>
      <Button 
      colorScheme={"red"} 
      variant={"outline"}
      onClick={()=>{
        setTheme(theme ==="light" ? "dark": "light");
      }}
      >
        Click me
        </Button>
    </div>
  );
}

export default App;
