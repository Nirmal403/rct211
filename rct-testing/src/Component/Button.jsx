import React from 'react';
import "./Button.css"

const Button = ({children, colorScheme, variant,
 onClick}) => {
  return (
    <button 
    data-testid="customButton"
    className={`button ${colorScheme || 'blue'} 
    ${variant || "ghost"}`}
    onClick={onClick}
    >
        {children}
    </button>
  )
}

export default Button