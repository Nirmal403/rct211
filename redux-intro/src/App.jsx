import "./App.css";
import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import axios from "axios";
import Counter from "./Component/Counter";
import {Todos} from "./Component/Todos"
import { loginSuccess, loginFailure , loginRequest} from "./Redux/AuthReducer/action";
function App() {
  const [useremail, setUserEmail] = useState("");
  const [password, setPassword] = useState("");
  const dispatch = useDispatch();
  const isAuth =useSelector((store)=>store.AuthReducer.isAuth);

  const handleLogin = () => {
    if (useremail) {
      const payload = {
        email: useremail,
        password,
      };
      dispatch(loginRequest());
      axios
        .post("https://reqres.in/api/login", payload)
        .then((r) =>{
          console.log(r.data);
            return dispatch(loginSuccess(r.data));
        })
        .catch((e) => dispatch(loginFailure(e)));
    }
  };
  return (
    <div className="App">
      <Counter />
      <div>
        <input
          placeholder="email"
          type="email"
          value={useremail}
          onChange={(e) => {
            setUserEmail(e.target.value);
          }}
        />
        <input
          placeholder="password"
          type="password"
          value={password}
          onChange={(e) => {
            setPassword(e.target.value);
          }}
        />
        <button onClick={handleLogin}>LOGIN</button>
      </div>

      {/* <TodoInput /> */}
      {isAuth && <Todos />}
    </div>
  );
}

export default App;
