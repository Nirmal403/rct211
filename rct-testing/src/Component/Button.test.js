import { render, screen, fireEvent } from "@testing-library/react";
import Button from "./Button";
import App from "../App";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";

describe("Testing the custom Button", () => {
  it("Should be present in the DOM", () => {
    render(<Button>Click Me</Button>);
    // screen.debug();
    let button = screen.getByText("Click Me");
    expect(button).toBeInTheDocument();
  });

  it("Should have a button with text click me", () => {
    render(<App></App>);
    // screen.debug();
    let button = screen.getByTestId("customButton");
    expect(button).toBeInTheDocument();
  });
  it("Should render without any children", () => {
    render(<Button></Button>);
    let button = screen.getByTestId("customButton");
    expect(button).toBeEmptyDOMElement();
  });
  it("Should toggle the theme", () => {
    render(<App></App>);
    let element = screen.getByText("Theme is light");
    expect(element).toHaveTextContent("light");

    let button = screen.getByTestId("customButton");
    userEvent.click(button);
    expect(element).toHaveTextContent("dark");

    userEvent.click(button);
    expect(element).toHaveTextContent("light");
  });

  it("Should call the given function", () => {
    const mockFunc = jest.fn();
    render(<Button onClick={mockFunc}>Click Me</Button>);

    const button = screen.getByTestId("customButton");
    fireEvent.click(button);
    fireEvent.click(button);

    expect(mockFunc).toBeCalledTimes(2);
  });

  it("Should match the snapshot", () => {
    const tree = renderer.create(
        <Button colorScheme={"green"}
         variant={"outline"}>
          Increment Counter
        </Button>
      )
      .toJSON();    
    expect(tree).toMatchSnapshot();
  });
});