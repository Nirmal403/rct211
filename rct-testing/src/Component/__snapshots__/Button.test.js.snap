// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the custom Button Should match the snapshot 1`] = `
<button
  className="button green 
    outline"
  data-testid="customButton"
>
  Increment Counter
</button>
`;
